### PR TITLE
feat: pass agent: ci in maestro in upload request when known CI env var is set

### PR DESCRIPTION
**Pull request by axelniklasson**

### CODE REVIEW

File: maestro-cli/src/main/java/maestro/cli/api/ApiClient.kt [hunk 1]
```diff
@@ -86,6 +89,17 @@ class ApiClient(
         }
     }
 
+    private fun getAgent(): String {
+        for (ciVar in knownCIEnvVars) {
+            try {
+                val value = System.getenv(ciVar)
+                if (value.isNotEmpty()) return "ci"
```
Comment (Leland-Takamine) [inline maestro-cli/src/main/java/maestro/cli/api/ApiClient.kt:96]: What if we have `CI=false`?

File: maestro-cli/src/main/java/maestro/cli/api/ApiClient.kt [hunk 2]
```diff
@@ -34,6 +35,8 @@ class ApiClient(
         .protocols(listOf(Protocol.HTTP_1_1))
         .build()
 
+    private val knownCIEnvVars = listOf("CI", "JENKINS_HOME")
```
Comment (dmitry-zaitsev) [inline maestro-cli/src/main/java/maestro/cli/api/ApiClient.kt:38]: What about Bitrise?

Comment (dmitry-zaitsev) [reply maestro-cli/src/main/java/maestro/cli/api/ApiClient.kt:38]: Also Circle CI

File: maestro-cli/src/main/java/maestro/cli/api/ApiClient.kt (hunk 1 above)
Comment (axelniklasson) [reply maestro-cli/src/main/java/maestro/cli/api/ApiClient.kt:96]: Good call, didn't think about that case. Do you think it makes more sense to consider it being ci iff `CI=true` or should we consider it being ci if `CI != false`?

Comment (Leland-Takamine) [reply maestro-cli/src/main/java/maestro/cli/api/ApiClient.kt:96]: That and also if set to "0"

File: maestro-cli/src/main/java/maestro/cli/api/ApiClient.kt (hunk 2 above)
Comment (axelniklasson) [reply maestro-cli/src/main/java/maestro/cli/api/ApiClient.kt:38]: According to their [docs](https://circleci.com/docs/variables/#built-in-environment-variables) they set `CI=true` so we should be good. realized that Bitrise only sets `CI=true` when running the CLI in continuous integration mode so will update to also check for the `BITRISE_IO` variable

**Comment by axelniklasson** [comment]
@dmitry-zaitsev @Leland-Takamine comments are addressed, let me know what you think!

**Review by Leland-Takamine — APPROVED** [review]

**Comment by dmitry-zaitsev** [comment]
@axelniklasson what about Circle CI?

**Comment by axelniklasson** [comment]
> @axelniklasson what about Circle CI?

@dmitry-zaitsev seems GitHub scrapped that comment but it is handled through checking for `CI=true` as that is a [default env variable](https://circleci.com/docs/variables/#built-in-environment-variables) in Circle CI

**Review by dmitry-zaitsev — APPROVED** [review]

**Pull request merged**